zulip/zulip/__init__.py

```diff
@@ -44,6 +45,20 @@
 
 API_VERSTRING = "v1/"
 
+EditPropagateMode = Literal["change_one", "change_all", "change_later"]
+
+EmojiType = Literal["realm_emoji", "unicode_emoji", "zulip_extra_emoji"]
+
+# Limited flags which could be present in update_message_flag events.
+ModifiableMessageFlag = Literal["read", "starred", "collapsed"]
+MessageFlag = Literal[
+    ModifiableMessageFlag,
+    "mentioned",
```

Comment:
For each of these, I'd encourage having a one- or two-line comment mentioning what part of the API it's for. Just a link to the relevant bit of API docs may be enough -- that way there's no need to try to duplicate any prose describing the semantics.

Reply:
These Literal types look a bit vague regarding how exactly the users are supposed to use them. For update_message_flags specifically, do you think that defining a TypedDict making use of these types will be a good idea?

Reply:
I referred to the commenting aspect in another response above.

Currently these types aren't directly used in the API, at least via other types, so we'd have to use TypedDicts or migrate the API to allow alternative function signatures.